editor/src/components/canvas/controls/select-mode/flex-gap-control.tsx

```diff
@@ -305,3 +377,46 @@ function startInteraction(
     ])
   }
 }
+
+interface NumberInputProps {
+  id: string
+  initialValue: number
+  onChange: (_: number) => void
+}
+
+const NumberInput = React.memo<NumberInputProps>((props) => {
+  const { onChange } = props
+  const [currentValue, setCurrentValue] = React.useState<number>(props.initialValue)
+  const onValueChange = React.useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
+    event.preventDefault()
+    event.stopPropagation()
+
+    const {
+      target: { value },
+    } = event
+    const valueNumeric = parseFloat(value)
+    setCurrentValue(valueNumeric)
+  }, [])
+
+  const onKeyDown = React.useCallback(
+    (e: React.KeyboardEvent<HTMLInputElement>) => {
+      if (e.code === 'Enter') {
+        onChange(currentValue)
+      }
+    },
+    [currentValue, onChange],
+  )
+
+  return (
+    <>
+      <input
+        onKeyDown={onKeyDown}
```

Comment:
If this input gets created / destroyed, you could add an `autofocus` here - there's a good chance it'll work as expected, so you can click & type.